packages/liveblocks-react/src/room.tsx

```diff
@@ -414,6 +414,20 @@ export function createRoomContext<
       void getThreadsUpdates(room.id);
     }, [room.id]);
 
+    /**
+     * Subscribe to the 'online' event to fetch threads/notifications updates when the browser goes back online.
+     */
+    React.useEffect(() => {
+      function handleIsOnline() {
+        void getThreadsUpdates(room.id);
+      }
+
+      window.addEventListener("online", handleIsOnline);
+      return () => {
+        window.removeEventListener("online", handleIsOnline);
+      };
+    }, [room.id]);
```

Comment:
Small suggestion: this effect could be combined with the one above, `handleIsOnline` could be called instantly (and renamed to something more generic like `getThreads`) in addition to being tied to the `"online"` event, and the effect above could be removed.

Reply:
I wanted to have a clear separation of what these `useEffect` callbacks do. The first one fetches thread updates when room id changes. The second one (this one) listens to browser going online and performs some tasks (fetching thread updates). In future, we may want to replace this with updating the poller's interval when browser comes back online, so I think the separation might be helpful even though it is a little verbose currently. What do you think?